### DISCUSSION TITLE
HiDPI instructions cause apps to get scaled twice

### DISCUSSION THREAD

**Discussion by steakscience** [hyprwm/Hyprland#1254] [Q&A]
Hi everyone :)

I followed the HiDPI instructions on the wiki, so now I am using a patched wlroots, etc

I have `GDK_SCALE=2` in my Hyprland wrapper and xprop stuff in my exec-once

This works in the sense that Xwayland apps are crisp, however, they are 2x the size they are supposed to be.

Does anyone have any suggestions?

**Comment by vaxerski** [comment]
@fufexan

**Comment by fufexan** [comment]
If they are not crisp, then the patches haven't been applied properly.

**Reply by steakscience** [reply]
Whoops sorry I had a typo. The apps are sharp, but they are scaled twice

**Reply by steakscience** [reply]
After following all the instructions, are apps supposed to be super large (scaled twice?) or did I do something wrong?

**Reply by fufexan** [reply]
They're supposed to have the same dimensions as wayland-native programs at a given scale, and be sharp.

**Reply by steakscience** [reply]
If my Xwayland apps are twice the size, what am I doing wrong?

Currently I have:

- hyprland config monitor set to 2x scale
- export GDK_SCALE=2 in my wrapper

**Reply by fufexan** [reply]
Does unsetting GDK_SCALE fix the double sizing?

**Reply by steakscience** [reply]
Yes, if I don't set GDK_SCALE apps are properly sized, however, they will be blurry.

**Reply by steakscience** [reply]
@fufexan I'm still running into this issue, but I'm not sure if my config is wrong, or this is how it works usually

**Reply by fufexan** [reply]
Can you give me some apps as an example, so I can try to reproduce this?

**Reply by steakscience** [reply]
Sure! Discord, Chrome, Steam

**Reply by fufexan** [reply]
Steam is the only app of the three which can only be run in XWayland. The other two can be run in Wayland mode with the `--ozone-platform=auto` flags. Discord may not work unless it's Canary or Webcord.

For Steam, I do not notice any issue when running it. `GDK_SCALE` is set to 2, and it opens with the size of a normal Wayland application.

**Reply by steakscience** [reply]
I'm so confused why it's different. Your monitor config in hypr config is also set to 2 scaling right?

**Reply by fufexan** [reply]
I don't have it set in `hyprland.conf`, I have it set using kanshi. Should not make any difference though, as it's still 2.

**Reply by steakscience** [reply]
I guess it is related to this: I am using patched packages instead of patching them myself:
- [hyprland-hidpi-xprop-git](https://aur.archlinux.org/packages/hyprland-hidpi-xprop-git)
- [xorg-xwayland-hidpi-xprop](https://aur.archlinux.org/packages/xorg-xwayland-hidpi-xprop)
- [wlroots-hidpi-xprop-git](https://aur.archlinux.org/packages/wlroots-hidpi-xprop-git)

**Reply by fufexan** [reply]
That could be an issue. May be a mismatched wlroots version, or something along those lines.

Try patching it by hand and see if it helps.

**Reply by pashtet04** [reply]
I removed this packages with break dependency:
```
pacman -Rdd xorg-xwayland-hidpi-xprop
```
And install this one from AUR:
```
paru hyprland-hidpi-xprop-git
paru xorg-xwayland-hidpi-xprop-git
```

xorg-xwayland-hidpi-xprop-git in AUR already with patch -- https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/733

Seems scaling is working for me now correctly

**Reply by pashtet04** [reply]
How to fix dependency permanently? It breaks on every update
